src/main/java/net/dzikoysk/funnyguilds/listener/region/BlockPlace.java

```diff
@@ -3,35 +3,87 @@
     private final PluginConfiguration config = FunnyGuilds.getInstance().getPluginConfiguration();
 
-    @EventHandler
-    public void onPlace(BlockPlaceEvent e) {
-        if (ProtectionSystem.build(e.getPlayer(), e.getBlock().getLocation(), true)) {
-            if (config.buggedBlocks && !config.buggedBlocksExclude.contains(e.getBlock().getType())) {
-                ItemStack returnItem = e.getItemInHand().clone();
-                returnItem.setAmount(1);
-                Bukkit.getScheduler().runTaskLater(FunnyGuilds.getInstance(), () -> {
-                    e.getBlockReplacedState().update(true);
-
-                    if (config.buggedBlockReturn) {
-                        e.getPlayer().getInventory().addItem(returnItem);
-                    }
-                }, config.buggedBlocksTimer);
-
-                return;
+    @EventHandler(priority = EventPriority.LOWEST)
+    public void onPlace(BlockPlaceEvent event) {
+        Player player = event.getPlayer();
+        Block block = event.getBlock();
+        Material type = block.getType();
+
+        Location blockLocation = block.getLocation();
+        if (!ProtectionSystem.build(player, blockLocation, true)) {
+            return;
+        }
+
+        // always cancel to prevent breaking other protection
+        // plugins or plugins using BlockPlaceEvent (eg. special ability blocks)
+        event.setCancelled(true);
+
+        // disabled bugged-blocks or blacklisted item
+        if (!this.config.buggedBlocks || this.config.buggedBlocksExclude.contains(type)) {
+            return;
+        }
+
+        // remove one item from the player
+        ItemStack itemInHand = event.getItemInHand();
+        if ((itemInHand.getAmount() - 1) == 0) {
+            // wondering why? because bukkit and you probably don't want dupe glitches
+            if (Reflections.USE_PRE_9_METHODS) {
+                player.setItemInHand(null);
+            } else {
+                itemInHand.setAmount(0);
             }
+        } else {
+            itemInHand.setAmount(itemInHand.getAmount() - 1);
+        }
 
-            e.setCancelled(true);
+        // if the player is standing on the placed block add some velocity to prevent glitching
+        // side effect: velocity with +y0.4 is like equivalent to jumping while building, just hold right click, that's real fun!
+        Location playerLocation = player.getLocation();
+        boolean sameColumn = (playerLocation.getBlockX() == blockLocation.getBlockX()) && (playerLocation.getBlockZ() == blockLocation.getBlockZ());
+        double distanceUp = (playerLocation.getY() - blockLocation.getBlockY());
+        boolean upToTwoBlocks = (distanceUp > 0) && (distanceUp <= 2);
+        if (sameColumn && upToTwoBlocks) {
+            player.setVelocity(new Vector(0, 0.4, 0));
         }
+
+        // delay, because we cannot do {@link Block#setType(Material)} immediately
+        Bukkit.getScheduler().runTaskLater(FunnyGuilds.getInstance(), () -> {
```

Comment:
Instead of using `BukkitScheduler#runTaskLater` with zero delay, we could use `BukkitScheduler#runTask`, as it has the same effect.